scripts/dev_setup.sh

```diff
@@ -415,19 +415,29 @@ function install_toolchain {
 }
 
 function install_rustup_components_and_nightly {
+    echo "Printing the rustup version and toolchain list"
+    rustup --version
+    rustup show
+    rustup toolchain list -v
+
     echo "Updating rustup and installing rustfmt & clippy"
     rustup update
     rustup component add rustfmt
     rustup component add clippy
 
     # We require nightly for strict rust formatting
     echo "Installing the nightly toolchain and rustfmt nightly"
-
-    # TODO: make the nightly error handling more robust
     if ! rustup toolchain install nightly
     then
-      echo "Failed to install the nightly toolchain using rustup."
+      # TODO: remove this once we have an answer: https://github.com/rust-lang/rustup/issues/3390
+      echo "Failed to install the nightly toolchain using rustup! Falling back to an older build..."
```

Comment:
what happens if the older build fails also? is this a known working build?

Reply:
This really shouldn't happen (i.e., the build has already succeeded and been published online, so it shouldn't flake). If it does happen, then I think we have a much larger problem, e.g., `rustup` has stopped working altogether. In which case, we want things to fail so we can be notified immediately 😱